app/config/config.php

```diff
@@ -83,6 +83,24 @@
     ],
     'fragments'            => null,
     'http_method_override' => true,
+    'mailer'               => [
+        //'message_bus' => 'email.bus', // The message_bus option was introduced in Symfony 5.1.
+        'dsn'         => '%env(MAUTIC_MAILER_DNS)%',
+    ],
+    'messenger'            => [
+        'default_bus' => 'email.bus',
+        'buses'       => [
+            'email.bus' => null,
+        ],
+        'transports'  => [
+            'email_transport' => [
```

Comment:
Awesome.

I do not understand options hosts/path under queue > redis configuration.


We use setup of messenger something like 

 ```
  'dsn'            => 'amqp://xxx:xxx@redis.local:7777',
        'options'        => [
            'queues'     =>
                [
                    'queue' =>
                        [
                            'binding_keys' =>
                                [
                                    0 => 'xxx',
                                ],
                        ],
                ],
            'auto_setup' => true,
            'exchange'   =>
                [
                    'name'                        => 'email',
                    'type'                        => 'direct',
                    'default_publish_routing_key' => xxx',
                ],
        ]
```

Probaby we should allow modify it by local parameter with retry_strategy too 

```
      'transports'  => [
            'email_transport' => [
                'dsn' => '%env(MAUTIC_MESSENGER_EMAIL_TRANSPORT_DSN)%',
                'options' => $configParameterBag->get('messenger_email_transport_options'),
                'retry_strategy' => $configParameterBag->get('messenger_email_transport_retry_strategy'),
            ],
        ],
```

Reply:
@kuzmany Thank you for feedback.

> I do not understand options hosts/path under queue > redis configuration.

We have implementation https://symfony.com/doc/current/messenger.html#redis-transport for redis. So `host` is host for redis (like `redis.local`). Path is a stream name (default `messages`). Maybe better rename to `Stream name`.

As for me, ideally to have only one DSN string for all transports (Every transport has their own options and configuration). So options are also can be set using DSN. 
Good point, to move `retry_strategy` as additional parameters for messenger.

Reply:
> As for me, ideally to have only one DSN string for all transports

Yes, that would be unify solution. 

> So options are also can be set using DSN.

That's not true. Special options must be passed by options parameter, like example above.

Reply:
@kuzmany all the ideas you mentioned are implemented in #11180 11180